src/ray/core_worker/context.cc

```diff
@@ -239,9 +236,12 @@ void WorkerContext::SetCurrentActorId(const ActorID &actor_id) LOCKS_EXCLUDED(mu
   current_actor_id_ = actor_id;
 }
 
+void WorkerContext::SetTaskDepth(int64_t depth) { task_depth_ = depth; }
```

Comment:
I think this needs to be guarded by the mutex_?

Reply:
The caller (SetCurrentTask) already holds the lock, we could refactor it so SetCurrentTask calls the function separately, or have coreworker call SetTaskDepth

Reply:
Ah I see, thanks. The annotation you added seems best.